modules/service/src/main/scala/lucuma/odb/Config.scala

```diff
@@ -141,6 +148,49 @@ object Config {
 
   }
 
+  case class Aws(
+    accessKey:       NonEmptyString,
+    secretKey:       NonEmptyString,
+    basePath:        NonEmptyString,
+    bucketName:      BucketName,
+    fileUploadMaxMb: Int 
+  ) {
+    def fileKey(programId: Program.Id, fileName: NonEmptyString): FileKey = 
+      FileKey(NonEmptyString.unsafeFrom(s"$basePath/$programId/$fileName"))
+  }
+
+  object Aws {
+    private implicit val showPath: Show[Uri.Path] = Show.fromToString
+
+    private implicit val uriPath: ConfigDecoder[Uri, Uri.Path] = 
+      ConfigDecoder[Uri].mapOption("Path")(_.path.some)
+      
+    // The basePath must be nonEmpty
+    private implicit val pathNES: ConfigDecoder[Uri.Path, NonEmptyString] =
+      ConfigDecoder[Uri.Path].mapOption("NonEmptyPath"){ p =>
+        val str = p.segments.map(_.encoded).mkString("/")
+        NonEmptyString.from(str).toOption
+      }
+
+    private implicit val bucketName: ConfigDecoder[Uri, BucketName] = 
+      ConfigDecoder[Uri].mapOption("BucketName"){ uri => 
+        uri.host.flatMap{ h => 
+          h.value.split("\\.").headOption
+         }.flatMap { b =>
+          NonEmptyString.from(b).toOption
+          .map(BucketName.apply) 
+        } 
+      }
+
+    lazy val fromCiris: ConfigValue[Effect, Aws] = (
+      envOrProp("CLOUDCUBE_ACCESS_KEY_ID").as[NonEmptyString],
+      envOrProp("CLOUDCUBE_SECRET_ACCESS_KEY").as[NonEmptyString].redacted,
+      envOrProp("CLOUDCUBE_URL").as[Uri].as[Uri.Path].as[NonEmptyString],
+      envOrProp("CLOUDCUBE_URL").as[Uri].as[BucketName],
```

Comment:
CloudCube is the Heroku service we're using, I guess?

Reply:
Yep. And it sets these variables for us.